src/runtime/typemanager.cs

```diff
@@ -696,8 +696,21 @@ internal static void InitializeNativeCodePage()
             // to be executable.
             IMemoryMapper mapper = CreateMemoryMapper();
             int codeLength = NativeCode.Active.Code.Length;
-            NativeCodePage = mapper.MapWriteable(codeLength);
-            Marshal.Copy(NativeCode.Active.Code, 0, NativeCodePage, codeLength);
+
+            // Request 7 bytes more than required such that we can align to an 8 byte boundary on ARM
+            NativeCodePage = mapper.MapWriteable(codeLength + 7);
+
+            // Round to 8 byte boundary
+            var start = new IntPtr(((NativeCodePage.ToInt64() + 7) >> 3) << 3);
```

Comment:
`>> 3) << 3)` can be replaced with `& ~0x7L`
IMHO, even better to write a generic align function (and unit test it)